identity-account/src/account/account.rs

```diff
@@ -119,25 +88,58 @@ impl Account {
 
   /// Returns the did of the managed identity.
   pub fn did(&self) -> &IotaDID {
-    &self.did
+    self.document().did()
+  }
+
+  /// Return the latest state of the identity.
+  pub fn state(&self) -> &IdentityState {
+    &self.state
+  }
+
+  pub(crate) fn state_mut_unchecked(&mut self) -> &mut IdentityState {
+    &mut self.state
+  }
```

Comment:
I don't think we should mark this as `unsafe` as this function does not violate memory safety. Unsafe operations [are defined](https://doc.rust-lang.org/reference/unsafety.html) to be :

> Unsafe operations are those that can potentially violate the memory-safety guarantees of Rust's static semantics.

Reply:
Perhaps, but we already have functions like `IotaDocument::as_document_mut` that are marked as `unsafe` because they can break invariants and essentially corrupt data (but not memory) due to unchecked changes, much like `state_mut_unchecked`. 

I would still prefer removing `state_mut_unchecked`.

Reply:
I will remove it.

I also would like us to re-evaluate what methods we mark `unsafe`. I agree it should be restricted to those that can violate memory-safety. For others, `_unchecked` and warning documentation seems good enough.

Reply:
>we already have functions like IotaDocument::as_document_mut that are marked as unsafe because they can break invariants and essentially corrupt data (but not memory) due to unchecked changes, much like state_mut_unchecked.

Then we should consider changing those methods. The `unsafe` keyword is not meant to be used in this way.

Reply:
`IotaDocument::as_document_mut` should be marked as `unsafe` because it allows the user to make changes that would violate other `unsafe` casts in `IotaDocument`. See all the comments in `validate_core_document`.